libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -871,19 +871,43 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			solAssert(_memberAccess.annotation().type, "_memberAccess has no type");
 			if (auto funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type.get()))
 			{
-				if (funType->location() != FunctionType::Location::Internal)
-				{
-					_memberAccess.expression().accept(*this);
-					m_context << funType->externalIdentifier();
-				}
-				else
+				switch (funType->location())
 				{
+				case FunctionType::Location::Internal:
 					// We do not visit the expression here on purpose, because in the case of an
 					// internal library function call, this would push the library address forcing
 					// us to link against it although we actually do not need it.
-					auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration);
-					solAssert(!!function, "Function not found in member access");
-					m_context << m_context.functionEntryLabel(*function).pushTag();
+					if (auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+						m_context << m_context.functionEntryLabel(*function).pushTag();
+					else
+						solAssert(false, "function not found");
+					break;
+				case FunctionType::Location::Log0:
+				case FunctionType::Location::Log1:
+				case FunctionType::Location::Log2:
+				case FunctionType::Location::Log3:
+				case FunctionType::Location::Log4:
+				case FunctionType::Location::Event:
+					if (!dynamic_cast<EventDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+						solAssert(false, "event not found");
+					// no-op, because the parent node will do the job
+					break;
+				case FunctionType::Location::External:
```

Comment:
I'll check the documentation to see how these symbols should be interpreted.  I also add tests on the dubious locations to see if the type checker or the name resolving already complains about these.